backend/devices/bitbox02/device.go

```diff
@@ -53,28 +53,31 @@ type Device struct {
 func NewDevice(
 	deviceID string,
 	version *semver.SemVer,
-	product bitbox02common.Product,
+	product *bitbox02common.Product,
 	config firmware.ConfigInterface,
 	communication firmware.Communication,
 	opts ...firmware.DeviceOption,
 ) *Device {
 	productName := BitBox02ProductName
 	// BitBox02Plus is the internal code name for the BitBox Nova.
-	if product == bitbox02common.ProductBitBox02PlusBTCOnly ||
-		product == bitbox02common.ProductBitBox02PlusMulti {
+	if product != nil && (*product == bitbox02common.ProductBitBox02PlusBTCOnly ||
+		*product == bitbox02common.ProductBitBox02PlusMulti) {
 		productName = BitBox02NovaProductName
```

Comment:
this also has to happen if the product name is inferred in instead of passed directly